drake/solvers/MobyLCP.cpp

```diff
@@ -129,20 +129,20 @@ bool MobyLCPSolver::solve(OptimizationProblem& prog) const {
   // Restriction 3 could reasonably be relaxed to simply let unbound
   // variables sit at 0.
 
-  assert(prog.getGenericConstraints().empty());
-  assert(prog.getGenericObjectives().empty());
-  assert(prog.getAllLinearConstraints().empty());
-  assert(prog.getBoundingBoxConstraints().empty());
+  assert(prog.get_generic_constraints().empty());
+  assert(prog.get_generic_objectives().empty());
+  assert(prog.get_all_linearConstraints().empty());
```

Comment:
As a possible explanation... last time I checked, I couldn't find any CI builds that had assertions turned on.
Probably worth testing BUILD_TYPE=Debug by hand before merging this.